java/core/src/main/java/com/microsoft/bond/protocol/ProtocolWriter.java

```diff
@@ -13,6 +13,12 @@
 public interface ProtocolWriter {
 
     /**
+     * @return true if bonded fields in this protocol are marshaled; false if
+     * they are serialized in the same protocol
+     */
+    boolean usesMarshaledBonded();
```

Comment:
Is has [similar code in a different place](https://github.com/Microsoft/bond/blob/2bdf48c618d1fb61de5488af0ed5d6988fcacb7e/cs/src/core/expressions/ProtocolWriter.cs#L162-L173) due its architecture of generating expressions.